src/api.rs

```diff
@@ -6,12 +6,15 @@ use serde::{de, Deserialize, Serialize};
 use serde_json;
 use std::fmt;
 use std::io::Write;
+use std::ops::DerefMut;
 use std::process::{Command, Stdio};
+use std::sync::Arc;
+use std::sync::Mutex;
 use url::Url;
 use users::get_current_username;
 
 pub struct API {
-  token: Option<String>,
+  token: Arc<Mutex<Option<String>>>,
```

Comment:
Sure! The good meme is that Rust wants only one mutable reference to a particular piece of memory, which means that we can't run two methods which need to mutate `self` at the same time (a conundrum!). So instead, I made the methods take `&API` (before they were `&mut API`). The token stuff still needs to be able to write so it can cache the API key we get from kerberos, so that's where the mutex comes in. We lock the mutex and check if it's `Some`. If it is, we just use that token, otherwise we get a new token. It's important that we don't release the mutex until after we get the new token, hence the new `take_token` method I added.

Reply:
Thanks!